libs/plugins/stream-client/src/lib/core/models/unique-stage-name.ts

```diff
@@ -0,0 +1,40 @@
+import { reduce, toInteger } from 'lodash';
+import { normalizeTaskName } from '@flogo-web/lib-client/common';
+
+import { Item } from '../interfaces';
+
+export function uniqueStageName(stageName: string, ...stageDictionaries) {
```

Comment:
this should be a candidate to refactor later to re-use with flow, I think if we add a `getElementName` function as parameter to obtain an element's name we can make this generic:

```ts
function makeNameUnique<T>(
 name: string,
 getElementName: (e: T) => string,
...elements: Dictionary<T>
): string {}
```